tidy3d/components/data/monitor_data.py

```diff
@@ -133,7 +133,136 @@ def symmetry_expanded_copy(self) -> AbstractFieldData:
+            raise DataError("Data must be 2D to get tangential dimensions.")
+        tangential_dims = ["x", "y", "z"]
+        tangential_dims.pop(zero_dims[0])
+
+        return tangential_dims
+
+    @property
+    def _tangential_fields(self) -> Dict[str, DataArray]:
+        """For a 2D monitor data, return a dictionary with only the tangential fields. Raise if any
+        of the tangential field components is missing."""
+        tan_dims = self._tangential_dims
+        field_components = ["E" + dim for dim in tan_dims] + ["H" + dim for dim in tan_dims]
+        for field in field_components:
+            if field not in self.field_components:
+                raise DataError(f"Tangential field component {field} is missing in data.")
+        normal_dim = "xyz"[self.monitor.size.index(0)]
+        return {field: getattr(self, field).squeeze(dim=normal_dim) for field in field_components}
+
+    @cached_property
+    def _plane_grid_boundaries(self) -> Tuple[ArrayLike[float, 1], ArrayLike[float, 1]]:
+        """For a 2D monitor data, return the boundaries of the in-plane grid from the stored field
+        coordinates."""
+        tan_dims = self._tangential_dims
+        tan_fields = self._tangential_fields
+        plane_bounds0 = tan_fields["E" + tan_dims[1]].coords[tan_dims[0]].values
+        plane_bounds1 = tan_fields["E" + tan_dims[0]].coords[tan_dims[1]].values
+        return plane_bounds0, plane_bounds1
+
+    @cached_property
+    def _diff_area(self) -> xr.DataArray:
+        """For a 2D monitor data, return the area of each cell in the plane, for use in numerical
+        integrations."""
+        bounds = self._plane_grid_boundaries
+        sizes = [bs[1:] - bs[:-1] for bs in bounds]
+        return xr.DataArray(np.outer(sizes[0], sizes[1]), dims=self._tangential_dims)
+
+    @property
+    def _centered_tangential_fields(self) -> Dict[str, DataArray]:
+        """For a 2D monitor data, get the tangential E and H fields colocated to the cell centers in
+        the 2D plane grid."""
+
+        # Tangential directions and fields
+        tan_dims = self._tangential_dims
+        tan_fields = self._tangential_fields
+
+        # Plane center coordinates
+        bounds = self._plane_grid_boundaries
+        centers = [(bs[1:] + bs[:-1]) / 2 for bs in bounds]
+
+        # Interpolate tangential field components to cell centers
+        interp_dict = dict(zip(tan_dims, centers))
+        centered_fields = {key: val.interp(**interp_dict) for key, val in tan_fields.items()}
+        return centered_fields
+
+    @property
+    def poynting(self) -> xr.DataArray:
+        """Time-averaged Poynting vector for frequency-domain data associated to a 2D monitor."""
+
+        # Tangential fields are ordered as E1, E2, H1, H2
+        tan_fields = self._centered_tangential_fields
+        field_list = list(tan_fields.values())
+        e_x_h_star = field_list[0] * field_list[3].conj() - field_list[1] * field_list[2].conj()
```

Comment:
why not just grab these from the dictionary?
```py
tan_fields['E1'] * tan_fields['H1'].conj() - tan_fields['E2'] * tan_fields['H2'].conj()
```

Reply:
Hm well they are still labeled e.g. `"Ex"` and `"Ez"` in the dictionary, which I think is more explicit still?

Reply:
Replaced everywhere using `self._tangential_dims`.

Reply:
looks good. Pretty nitpicky / optional but it might make it a little cleaner to unpack the `tan_dims`.
```py
tan_dims = self._tangential_dims
e_x_h_star = tan_fields["E" + tan_dims[0]] * tan_fields["H" + tan_dims[1]].conj()
e_x_h_star -= tan_fields["E" + tan_dims[1]] * tan_fields["H" + tan_dims[0]].conj()
```
to
```py
dim1, dim2 = self._tangential_dims
e_x_h_star = tan_fields["E" + dim1] * tan_fields["H" + dim2].conj()
e_x_h_star -= tan_fields["E" + dim2] * tan_fields["H" + dim1].conj()
```